homeassistant/components/sonos/media_player.py

```diff
@@ -427,7 +429,10 @@ def device_info(self):
     @soco_coordinator
     def state(self):
         """Return the state of the entity."""
-        if self._status in ("PAUSED_PLAYBACK", "STOPPED"):
+        if self._media_title is not None and self._status in (
```

Comment:
I am not sure that it can fail but honestly, this makes me feel odd. It seems to conflate concerns. Maybe we paused some untagged media? Even if that cannot happen with Sonos (I am unsure), it seems like a situation the frontend should be able to handle?

Reply:
I consider it a bug in Home Assistant if a media player is claiming to have media loaded (being in a paused/playing) and it cannot describe that media and it normally can.

Reply:
Okay, in that case we have a bug in `update_media_linein()` :)

Maybe it would look less out of place as a guard.

Reply:
Oh sorry, with MP3 I meant one from a local server. You're right linein is different. Do we know the transport info if it's linein / TV ?

Reply:
With a guard, do you mean that you want to see the if be inside the other if ?

Reply:
Line-in currently has
```
        self._media_artist = source
```
that we can just change to
```
        self._media_title = source
```
and it is hard-coded as "TV" or "Line-in".

Reply:
I meant
```
    if self._media_title is None:
        return STATE_IDLE
```
but don't change it just for me, I'm fine with whatever :)

Reply:
done and done!